### DISCUSSION TITLE
[V3] Ability to customize registration process

### DISCUSSION THREAD

**Discussion by billmn** [filamentphp/filament#7247] [Ideas]
What about add ability to customize user registration process like adding `beforeCreate` and `afterCreate` methods in Filament\Pages\Auth\Register class?

This would allow to customize the registration process without having to override the entire `register` method.

The `beforeCreate` probably needs the current form state, while `afterCreate` needs the form state and the created user model.

**Comment by danharrin** [comment]
Sure, I'm happy to add hooks to any of the new auth pages

**Comment by billmn** [comment]
Hey @danharrin,
do you prefer an issue for this FR?

Regarding the registration page, it might be useful to be able to choose whether or not to log in automatically.

**Reply by danharrin** [reply]
To clarify, I am happy to merge PRs that add new render hooks. But I am not adding the code myself

**Reply by billmn** [reply]
Oh forgive me I had misunderstood.
No problem, as soon as I can I will send you a PR.

Thanks
